src/NServiceBus.Metrics/ProbeBuilders/SignalProbeBuilder.cs

```diff
@@ -0,0 +1,30 @@
+using System;
+using System.Reflection;
+
+abstract class SignalProbeBuilder
+{
+    protected abstract void WireUp(SignalProbe probe);
+
+    public SignalProbe Build()
+    {
+        var probe = GetProbe();
+
+        WireUp(probe);
+
+        return probe;
+    }
+
+    SignalProbe GetProbe()
+    {
+        var attribute = GetType().GetCustomAttribute<ProbePropertiesAttribute>();
+
+        if (attribute == null)
+        {
+            var exceptionMessage = $"The type '{GetType()}' is not annotated with required '{typeof(ProbePropertiesAttribute).Name}'. This attribute has to be added to provide necessary metadata for the probe.";
```

Comment:
Me neither. Both `DurationProbeBuilder` and `SignalProbeBuilder` look 100% alike, if I'm not mistaken. Perhaps you can create a generic class to eliminate a class?